GitCommands/Git/GitModule.cs

```diff
@@ -3062,16 +3062,20 @@ public string OpenWithDifftool(string filename, string oldFileName = "", string
             return output;
         }
 
-        public string RevParse(string revisionExpression)
+        [CanBeNull]
+        public ObjectId RevParse(string revisionExpression)
         {
             string revparseCommand = string.Format("rev-parse \"{0}~0\"", revisionExpression);
             var result = RunGitCmdResult(revparseCommand);
-            return result.ExitCode == 0 ? result.StdOutput.Split('\n')[0] : "";
+            return result.ExitCode == 0
+                ? ObjectId.Parse(result.StdOutput, offset: 0)
+                : null;
         }
 
-        public string GetMergeBase(string a, string b)
+        [NotNull]
+        public ObjectId GetMergeBase(string a, string b)
         {
-            return RunGitCmd("merge-base " + a + " " + b).TrimEnd();
+            return ObjectId.Parse(RunGitCmd("merge-base " + a + " " + b), offset: 0);
```

Comment:
`Parse` overloads that take an offset allow parsing from within an existing string without having to allocate a substring. Hence we don't need to trim.